src/Test/Perf/runner.csx

```diff
@@ -8,114 +8,48 @@ using System.Collections.Generic;
 using System.IO;
 using System;
 
-InitUtilities();
-
-var testDirectory = Path.Combine(MyWorkingDirectory(), "Tests");
-
-var allResults = new List<Tuple<string, List<Tuple<int, string, object>>>>();
-var failed = false;
-
-var traceManager = TraceManagerFactory.GetTraceManager();
+var directoryInfo = new RelativeDirectory();
+var testDirectory = Path.Combine(directoryInfo.MyWorkingDirectory, "Tests");
 
 // Print message at startup
 Log("Starting Performance Test Run");
 Log("hash: " + StdoutFrom("git", "show --format=\"%h\" HEAD --").Split(new[] {"\r\n", "\r", "\n"}, StringSplitOptions.None)[0]);
 Log("time: " + DateTime.Now.ToString());
 
-traceManager.Setup();
+var testInstances = new List<dynamic>();
 
-for(int i = 0; i < traceManager.Iterations; ++ i)
+// Find all the tests from inside of the csx files.
+foreach (var script in GetAllCsxRecursive(testDirectory))
 {
-    traceManager.Start();
+    var scriptName = Path.GetFileNameWithoutExtension(script);
+    Log($"Collecting tests from {scriptName}");
+    var tests = (object[]) (await RunFile(script)).ReturnValue;
```

Comment:
FYI, I've been working on a small refactoring that moves all the non-scripty things into an assembly that we build and #r, so this will get done.